sdk/communication/AzureCommunicationChat/Source/Signalling/TrouterUtils.swift

```diff
@@ -0,0 +1,372 @@
+//
+//  TrouterUtils.swift
+//  AzureCommunicationSignaling
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import AzureCommunication
+import AzureCore
+import Foundation
+import TrouterModulePrivate
+
+func toEventPayload(request: TrouterRequest, chatEventId: ChatEventId) -> Any? {
+    do {
+        switch chatEventId {
+        case ChatEventId.chatMessageReceived:
+            return try toChatMessageReceivedEvent(request: request)
+        case ChatEventId.typingIndicatorReceived:
+            return try toTypingIndicatorReceivedEvent(request: request)
+        case ChatEventId.readReceiptReceived:
+            return try toReadReceiptReceivedEvent(request: request)
+        case ChatEventId.chatMessageEdited:
+            return try toChatMessageEditedEvent(request: request)
+        case ChatEventId.chatMessageDeleted:
+            return try toChatMessageDeletedEvent(request: request)
+        case ChatEventId.chatThreadCreated:
+            return try toChatThreadCreatedEvent(request: request)
+        case ChatEventId.chatThreadPropertiesUpdated:
+            return try toChatThreadPropertiesUpdatedEvent(request: request)
+        case ChatEventId.chatThreadDeleted:
+            return try toChatThreadDeletedEvent(request: request)
+        case ChatEventId.participantsAdded:
+            return try toParticipantsAddedEvent(request: request)
+        case ChatEventId.participantsRemoved:
+            return try toParticipantsRemovedEvent(request: request)
+        }
+    } catch {
+        return nil
+    }
+}
+
+func toChatMessageReceivedEvent(request: TrouterRequest) throws -> ChatMessageReceivedEvent {
+    do {
+        let requestJsonData = request.body.data(using: .utf8)!
+        let messageReceivedPayload: MessageReceivedPayload = try JSONDecoder()
+            .decode(MessageReceivedPayload.self, from: requestJsonData)
+
+        let chatMessageReceivedEvent =
+            ChatMessageReceivedEvent(
+                threadId: messageReceivedPayload.groupId,
+                sender: CommunicationUserIdentifier(messageReceivedPayload.senderId),
```

Comment:
Am updating all of these, realized this is incorrect assumption that it is always CommunicationUserIdentifier.